docs/book/user-guide/advanced-guide/data-versioning-and-lineage.md

```diff
@@ -1,25 +1,73 @@
-# Data versioning and lineage
+---
+description: How data is managed in ZenML.
+---
 
-In this lesson we will learn about one of the coolest features of ML pipelines: automated artifact versioning and tracking. This will give us tremendous insights into how exactly each of our models was created. Furthermore, it enables artifact caching, allowing us to switch out parts of our ML pipelines without having to rerun any previous steps.
+# Data versioning and artifact management in ZenML
 
-First, if you have not done so already, run the following cell to install ZenML and it's sklearn integration.
+ZenML seamlessly integrates data versioning and lineage into its core functionality. When a pipeline is executed, each run generates artifacts that are automatically tracked and managed. One can easily view the entire lineage of how artifacts are created and interact with them through the [post-execution workflow](../starter-guide/fetch-runs-after-execution.md). ZenML's artifact management, caching, lineage tracking, and visualization capabilities can help gain valuable insights, streamline the experimentation process, and ensure the reproducibility and reliability of machine learning workflows.
 
-## Artifact storage
+## Artifact Versioning, Caching, and Lineage
 
-You might now wonder how our ML pipelines can keep track of which artifacts changed and which did not. This requires several additional MLOps components that you would typically have to set up and configure yourself. Luckily, ZenML automatically set this up for us.
+Each time a ZenML pipeline runs, the system first checks if there have been any changes in the inputs, outputs, parameters, or configuration of the pipeline steps. Each step in a run gets a new directory:
 
-Under the hood, all the artifacts in our ML pipeline are automatically stored in an Artifact Store. By default, this is simply a place in your local file system, but we could also configure ZenML to store this data in a cloud bucket like Amazon S3 or any other place instead. We will see this in more detail when we migrate our MLOps stack to the cloud in a later chapter.
+![Visualizing artifacts](../../.gitbook/assets/zenml_artifact_store_underthehood_1.png)
 
-## Orchestrators
+If a step is new or has been modified, ZenML creates a new directory structure in the [Artifact Store](../component-galery/artifact-stores/README.md) with a unique ID and stores the data using the appropriate materializers in this directory.
```

Comment:
... new directory structure ... - i feel this is too detailed

Reply:
Really? i think its useful

Reply:
looking at it now I am also not too sure what my issue with this was. Maybe the wording? 

If a step is new or has been modified, ZenML creates a new **folder** in the ...

Maybe using the word folder it sounds better already

Reply:
Ok ill see